ideas/314-tribute-to-talk/specs.md

```diff
@@ -0,0 +1,90 @@
+# Tribute to Talk Specifications
+
+There are two layers to TtT:
+- Anti-spam 1:1 economic filtering. This is what we are building in the MVP.
+- Monetization of a user’s time & attention with added features such as escrow, conversation time-out/limitations, etc.
+
+The MVP will use a simple registry contract. For future iterations we will need use additional contracts. Research on contract options: https://notes.status.im/SGNQNVEATuChD7xXHybyWA 
+
+## MVP
+_Anti-spam 1:1 economic filtering_
+
+_Goals:_ Deliver on white paper promise to SNT holders. Boost SNT utility. Provide basis for personal time/attention monetization through anti-spam 1:1 economic filtering.
+
+### User Stories
+
+_Summary:_ Anyone can set a “tribute to talk.” Others can pay this tribute via a normal wallet transaction in order to open up a conversation with that individual. Once they’ve paid the user tribute, they are added to their contacts. There is currently no way for a contact to be removed. This might be a blocker that we should tackle first.
+
+_User A_
+- As a user who values privacy, I would like to set a minimum required tribute to talk to me in order to prevent 1:1 spam. 
+- I would like to change the tribute required to talk to me.
+- I would like to remove the tribute required to talk to me. 
+
+_A flows_<br>
+<img src="https://github.com/status-im/swarms/blob/master/ideas/314-tribute-to-talk/mvp-user-a.png" width=350>
+
+_User B_
+- As a user looking to make new contacts, I would like to pay tribute so that I can message User A.
+
+_B flows_<br>
+<img src="https://github.com/status-im/swarms/blob/master/ideas/314-tribute-to-talk/mvp-user-b.png" width=350> 
+
+### Feature Branding
+
+Maintaining “Tribute to Talk” for naming to draw connection to white paper.
+
+Treating this MVP as an early version of white paper Tribute to Talk (rather than separate feature).
+
+### Open Questions
+- Identifying TtT transactions in wallet history. Can we customize this?
+- Is it a full wallet flow when setting/editing/removing a Tribute? Do we want to minimize this?
+- How to handle `Learn More` screen (provided for in designs)
+- How to mitigate the awkwardness of trying to contact someone you know, when they have a tribute set? Options: (1) Contact request feature (WIP) to circumvent TtT; (2) Refund TtT fee for someone I know. 
+
+### Dependencies/Blockers
+- `Remove from contacts` option would be helpful to remove people. Core team will not be able to work on this soon; if wanted, chat can own.
+- ENS names should be supported by default in chat to boost feature usability. Why would I pay tribute to an anon name?
+
+### Designs
+[Figma link](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=1319%3A13403 )
+ 
+### Measurement & Tracking
```

Comment:
Maybe missed it, but how are we going to track the tributes? App-wise or contract-wise?

Reply:
@siphiuel tribute settings will be tracked in the contract, but tribute payment will be handled via a regular wallet transaction (no interaction with contract).